webapp/loggedin/modules/analysis/processingChain/processingChainView.js

```diff
@@ -0,0 +1,70 @@
+import React, { useEffect } from 'react'
+import { connect } from 'react-redux'
+import * as R from 'ramda'
+
+import Survey from '../../../../../common/survey/survey'
+import ProcessingChain from '../../../../../common/analysis/processingChain'
+import ObjectUtils from '../../../../../common/objectUtils'
+
+import LabelsEditor from '../../../surveyViews/labelsEditor/labelsEditor'
+
+import { getUrlParam } from '../../../../utils/routerUtils'
+
+import * as SurveyState from '../../../../survey/surveyState'
+import * as ProcessingChainViewState from './processingChainViewState'
+
+import { fetchProcessingChain, putProcessingChainProp } from './actions'
+
+const ProcessingChainView = props => {
+
+  const {
+    surveyInfo, processingChainUuid, processingChain,
+    fetchProcessingChain, putProcessingChainProp,
+  } = props
+
+  useEffect(() => {
+    fetchProcessingChain(processingChainUuid)
+  }, [])
+
+  const onPropLabelsChange = key => labelItem => {
+    const labelsUpdated = R.pipe(
+      ObjectUtils.getProp(key),
+      R.assoc(labelItem.lang, labelItem.label),
+    )(processingChain)
+
+    putProcessingChainProp(key, labelsUpdated)
+  }
+
+  return processingChain
```

Comment:
can this be ```return processingChain && ```

Reply:
we have to return "null" when the processingChain has not been loaded yet